service_configuration_lib/spark_config.py

```diff
@@ -643,11 +643,14 @@ def get_resources_requested(spark_opts: Mapping[str, str]) -> Mapping[str, int]:
     num_gpus = int(spark_opts.get('spark.mesos.gpus.max', 0))
 
     executor_memory = parse_memory_string(spark_opts.get('spark.executor.memory', ''))
+    requested_memory = spark_opts.get(
+        'spark.executor.memoryOverhead',
+    ) or spark_opts.get('spark.mesos.executor.memoryOverhead')
```

Comment:
We should also include `spark.kubernetes.memoryOverheadFactor` in the check too.
It seems to me by applying the standard definition of configuration overriding by child class, that `spark.mesos.executor.memoryOverhead` and `spark.kubernetes.memoryOverheadFactor` would override `spark.executor.memoryOverhead`.
To be on safe-side for resource allocation, should we take maximum of these 3?